### PR TITLE
Init command improvements

### PR DESCRIPTION
**Pull request by rtluckie**
- add config writing to init command
- more cleanup and reorg
- allow ability to run multiple KDK containers simultaneously

### CODE REVIEW

File: cmd/ssh.go [hunk 1]
```diff
@@ -31,9 +32,8 @@ var sshCmd = &cobra.Command{
 		logger := logrus.New().WithField("command", "ssh")
 
 		connectionString := viper.Get("docker.environment.KDK_USERNAME").(string) + "@localhost"
-
 		logger.Info("Connecting to KDK container")
-		sh.Command("ssh", connectionString, "-A", "-p", "2022", "-i", "~/.kdk/ssh/id_rsa", "-o", "StrictHostKeyChecking=no", "-o", "UserKnownHostsFile=/dev/null").SetStdin(os.Stdin).Run()
+		sh.Command("ssh", connectionString, "-A", "-p", kdk.Port, "-i", "~/.kdk/ssh/id_rsa", "-o", "StrictHostKeyChecking=no", "-o", "UserKnownHostsFile=/dev/null").SetStdin(os.Stdin).Run()
```
Comment (dcwangmit01) [inline cmd/ssh.go]: Nit.  I would newline a few of these:
```
               session.Command("ssh",
                       viper.Get("docker.environment.KDK_USERNAME").(string) + "@localhost",
                       "-A",
                       "-p", "2022",
                       "-i", "~/.kdk/ssh/id_rsa",
                       "-o", "StrictHostKeyChecking=no",
                       "-o", "UserKnownHostsFile=/dev/null").SetStdin(os.Stdin).Run()
```

I'm old school and still respect 80 chars, i mean 140 chars.

File: internal/app/kdk/destroy.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	var containerIds []string
+
+	containers, err := dockerClient.ContainerList(ctx, types.ContainerListOptions{})
+
+	if err != nil {
+		logger.WithField("error", err).Fatal("Failed to list docker containers")
+	}
+	for _, container := range containers {
+		for _, name := range container.Names {
+			if strings.HasPrefix(name, "/kdk") {
```
Comment (dcwangmit01) [inline internal/app/kdk/destroy.go]: Will destroy kill ALL kdk's?  You showed me a demo where you could launch multiple kdk's.

Comment (rtluckie) [reply internal/app/kdk/destroy.go]: It will destroy all containers with `kdk` prefix.  Maybe we should add a `prefix string` arg to `Destroy`.

Comment (dcwangmit01) [reply internal/app/kdk/destroy.go]: You specifically name the kdk in the config.yaml.  Can we pick it out by name what to destroy?

Regardless, this is a should and not a must.   We can followup a pr later for multi-kdk support.

File: internal/app/kdk/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+func Snapshot(ctx context.Context, dockerClient *client.Client, logger logrus.Entry) error {
+
+	snapshotName := Name + "-" + strconv.Itoa(int(time.Now().UnixNano()))
```
Comment (dcwangmit01) [inline internal/app/kdk/snapshot.go]: Where does Name come from?  Is it a global?

File: internal/app/kdk/init.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	"os/user"
+	"path"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/cisco-sso/kdk/internal/pkg/utils"
+	"github.com/docker/docker/client"
+	"github.com/manifoldco/promptui"
+)
+
+var (
```
Comment (dcwangmit01) [inline internal/app/kdk/init.go]: This kind of looks like a struct, but it isn't :)

File: internal/app/kdk/snapshot.go (hunk 3 above)
Comment (rtluckie) [reply internal/app/kdk/snapshot.go]: https://github.com/cisco-sso/kdk/pull/30/files#diff-c9bb13bf6d2baafd0e7014870515300fR35

File: cmd/ssh.go (hunk 1 above)
Comment (dcwangmit01) [reply cmd/ssh.go]: I bet we could run this all through gofmt, and be done with the formatting.

File: internal/app/kdk/init.go (hunk 4 above)
Comment (rtluckie) [reply internal/app/kdk/init.go]: should probably be a struct.

File: internal/app/kdk/destroy.go (hunk 2 above)
Comment (dcwangmit01) [reply internal/app/kdk/destroy.go]: Perhaps  destroy can pattern match on the global "Name"

File: internal/app/kdk/snapshot.go (hunk 3 above)
Comment (rtluckie) [reply internal/app/kdk/snapshot.go]: var scope is limited to `package kdk`  _i think_

File: cmd/ssh.go (hunk 1 above)
Comment (rtluckie) [reply cmd/ssh.go]: I ran `gofmt -s -w cmd internal main.go` before opening PR.

File: internal/app/kdk/destroy.go (hunk 2 above)
Comment (rtluckie) [reply internal/app/kdk/destroy.go]: IDK... I'd like to be able to destroy all `kdk` containers regardless of config.

also, notice that user is prompted for confirmation before deletion

**Pull request merged**